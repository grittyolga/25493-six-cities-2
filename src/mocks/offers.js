
const Offers = [
  {
    mark: `Premium`,
    image: `img/apartment-01.jpg`,
    price: 120,
    bookmark: false,
    rating: 93,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    ratingNumber: 4.8,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
  {
    image: `img/room.jpg`,
    price: 80,
    bookmark: true,
    rating: 80,
    name: `Wood and stone place`,
    type: `Private room`,
    ratingNumber: 4,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
  {
    image: `img/apartment-02.jpg`,
    price: 132,
    bookmark: false,
    rating: 80,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    ratingNumber: 4,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
  {
    mark: `Premium`,
    image: `img/apartment-01.jpg`,
    price: 180,
    bookmark: false,
    rating: 100,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    ratingNumber: 5,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
];


export default Offers;
