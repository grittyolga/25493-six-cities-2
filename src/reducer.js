import Offers from "../src/mocks/offers";


const initialState = {
  city: Offers[0].city,
  cityOffers: Offers.filter((offer) => offer.city === Offers[0].city)
};

const ActionType = {
  CHANGE_CITY: `CHANGE_CITY`,
};


const ActionCreator = {
  changeCity: (city) => ({
    type: ActionType.CHANGE_CITY,
    payload: city,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return Object.assign({}, state, {
        city: action.payload,
        cityOffers: Offers.filter((offer) => offer.city === action.payload)
      });
  }
  return state;
};


export {
  ActionCreator,
  ActionType,
  reducer,
};
