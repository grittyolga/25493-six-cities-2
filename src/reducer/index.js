import {combineReducers} from "redux";
import {reducer as data} from "./data/data";
import {reducer as userstate} from "./userstate/userstate";
import NameSpace from "./name-spaces";

export default combineReducers({
  [NameSpace.DATA]: data,
  [NameSpace.USERSTATE]: userstate
});
