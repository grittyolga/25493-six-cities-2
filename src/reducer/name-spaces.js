const NameSpace = {
  DATA: `DATA`,
  USERSTATE: `USERSTATE`
};

export default NameSpace;
