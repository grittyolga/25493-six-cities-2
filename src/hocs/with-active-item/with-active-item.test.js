import React from "react";
import renderer from "react-test-renderer";
import offers from "../../mocks/offers.js";
import {Provider} from "react-redux";
import {createStore} from "redux";
import {ActionCreator, reducer} from "../../reducer";
import withActiveItem from "./with-active-item";
import Cardlist from "../../components/cardlist/cardlist";
import Citylist from "../../components/citylist/citylist";

function getCityList(cardOffers) {
  const cityList = Array.from(cardOffers, (x) => x.city);
  return Array.from(new Set(cityList));
}


it(`With-Active-Item hoc works`, () => {
  const store = createStore(
      reducer,
      window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
  );
  const CardlistWrapped = withActiveItem(Cardlist, `activeCard`, ActionCreator.changeActiveCard);
  const CitylistWrapped = withActiveItem(Citylist, `city`, ActionCreator.changeCity);
  const tree = renderer
    .create(<Provider store={store}>
      <CardlistWrapped
        cityOffers={offers.filter((offer) => offer.city === offers[0].city)}
      />
      <CitylistWrapped
        cityList = {getCityList(offers)}
      />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
