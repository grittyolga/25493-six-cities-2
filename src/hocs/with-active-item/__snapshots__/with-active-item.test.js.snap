// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With-Active-Item hoc works 1`] = `
Array [
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="cities__place-card place-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/apartment-01.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              120
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button  button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "[object Object]%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
          >
            Beautiful & luxurious apartment at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/apartment-01.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              215
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button  button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "[object Object]%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
          >
            Luxurious apartment
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="cities__place-card place-card"
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/apartment-small-04.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              66
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button  button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "[object Object]%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="#"
          >
            Nice apartment at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </div>,
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          href="#"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Cologne
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item "
          href="#"
          onClick={[Function]}
        >
          <span>
            Brussels
          </span>
        </a>
      </li>
    </ul>
  </section>,
]
`;
